File: bwl.Framework/src/AutoUI/RemoteElements/RemoteAutoTextbox.vb
Imports System.Drawing

Public Class RemoteAutoTextbox
    Inherits BaseRemoteElement

    Public Sub New()
        Me.New(New UIElementInfo("", ""))
    End Sub

    Public Sub New(info As UIElementInfo)
        MyBase.New(info)
        InitializeComponent()
        AddHandler info.Changed, AddressOf BaseInfoChanged
        BaseInfoChanged(info)
    End Sub

    Private Sub BaseInfoChanged(source As UIElementInfo)
        If InvokeRequired Then
            Me.Invoke(Sub() BaseInfoChanged(source))
        Else
            ElementCaption.Text = Info.Caption
            If Info.BackColor.A = 255 Then TextBox1.BackColor = Info.BackColor
            If Info.Width > 0 Then Me.Width = Info.Width
            If Info.Height > 0 Then Me.Height = Info.Height
        End If
    End Sub

    Public Overrides Sub ProcessData(dataname As String, data() As Byte)
        If dataname.ToLower = "text" Then
            Dim text = AutoUIByteCoding.GetString(data)
            Me.Invoke(Sub() If Me.TextBox1.Text <> text Then Me.TextBox1.Text = text)
        End If
    End Sub

    Private Sub TextBox1_KeyUp(sender As Object, e As KeyEventArgs) Handles TextBox1.KeyUp
        Send("text-changed", {TextBox1.Text})
    End Sub

    Private Sub TextBox1_TextChanged(sender As Object, e As EventArgs) Handles TextBox1.TextChanged

    End Sub

    Private Sub TextBox1_Click(sender As Object, e As EventArgs) Handles TextBox1.Click
        Send("click", {})

    End Sub

    Private Sub TextBox1_DoubleClick(sender As Object, e As EventArgs) Handles TextBox1.DoubleClick
        Send("double-click", {})

    End Sub

    Private Sub RemoteAutoTextbox_Load(sender As Object, e As EventArgs) Handles MyBase.Load

    End Sub
End Class

File: Framework/Framework/Bwl.Framework.Windows/AutoUI/RemoteElements/RemoteAutoTextbox.cs
using System.Runtime.Versioning;

namespace Bwl.Framework.Windows
{

    [SupportedOSPlatform("windows")]
    public partial class RemoteAutoTextbox : BaseRemoteElement
    {

        public RemoteAutoTextbox() : this(new UIElementInfo("", ""))
        {
        }

        public RemoteAutoTextbox(UIElementInfo info) : base(info)
        {
            InitializeComponent();
            info.Changed += BaseInfoChanged;
            BaseInfoChanged(info);
        }

        private void BaseInfoChanged(UIElementInfo source)
        {
            if (InvokeRequired)
            {
                Invoke(() => BaseInfoChanged(source));
            }
            else
            {
                ElementCaption.Text = Info.Caption;
                if (Info.BackColor.A == 255)
                    TextBox1.BackColor = GetColor(Info.BackColor);
                if (Info.ForeColor.A == 255)
                    TextBox1.ForeColor = GetColor(Info.ForeColor);
                if (Info.Width > 0)
                    Width = Info.Width;
                if (Info.Height > 0)
                    Height = Info.Height;
                try
                {
                    if (Info.ElemValue is not null)
                        TextBox1.Text = (string)Info.ElemValue;
                }
                catch (Exception ex)
                {
                }
            }
        }

        public override void ProcessData(string dataname, byte[] data)
        {
            if (dataname.ToLower() == "text")
            {
                string text = AutoUIByteCoding.GetString(data);
                Invoke(() => { if ((TextBox1.Text ?? "") != (text ?? "")) TextBox1.Text = text; });
            }
        }

        private void TextBox1_KeyUp(object sender, KeyEventArgs e)
        {
            Send("text-changed", new[] { TextBox1.Text });
        }

        private void TextBox1_TextChanged(object sender, EventArgs e)
        {

        }

        private void TextBox1_Click(object sender, EventArgs e)
        {
            Send("click", Array.Empty<object>());

        }

        private void TextBox1_DoubleClick(object sender, EventArgs e)
        {
            Send("double-click", Array.Empty<object>());

        }

        private void RemoteAutoTextbox_Load(object sender, EventArgs e)
        {

        }
    }
}